imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,208 @@
+"""Decommutate HIT CCSDS data and create L1a data products."""
+
+import logging
+from collections import defaultdict
+from dataclasses import fields
+from enum import IntEnum
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import decom, imap_module_directory, utils
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit import hit_cdf_attrs
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+
+logger = logging.getLogger(__name__)
+
+
+class HitAPID(IntEnum):
+    """
+    HIT APID Mappings.
+
+    Attributes
+    ----------
+    HIT_AUT : int
+        Autonomy
+    HIT_HSKP: int
+        Housekeeping
+    HIT_SCIENCE : int
+        Science
+    HIT_IALRT : int
+        I-ALiRT
+    HIT_MEMDUMP : int
+        Memory dump
+    """
+
+    HIT_AUT = 1250  # Autonomy
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+    HIT_MEMDUMP = 1255  # Memory dump
+
+
+def hit_l1a_data(packet_file: Path | str):
+    """
+    Process HIT L0 data into L1A data products.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    dict
+        List of file paths to CDF data product files
+    """
+    # Decom, sort, and group packets by apid
+    packets = decom_packets(packet_file)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    grouped_data = group_data(sorted_packets)
```

Comment:
Similar to Greg's comment, there is this `group_by_apid()` that you can use to group your data per apid.https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/e1234fe0f69ab76a3d3484114abb8f200b2e1989/imap_processing/utils.py#L39

Reply:
I use that global function in my custom group_data function to group the packets from space packet parser. However, the data need to be stored as data classes for further decommutation steps and so my function replaces the space packet parser packets with those data classes in the grouped_data dictionary it outputs. I could rename this function so that it isn't confused with the global group_by_apid function. Thoughts?

Reply:
Like previously mentioned, I do some dataclass oriented sorting (especially in MAG) if you want to take a look. Maybe there is shared work in that design that we can pull out.

Reply:
Oh ok. I should have look at function for more details. Yeah, I thought this function was doing same thing as utils function. Sounds good.